source/common/tracing/zipkin/zipkin_core_types.cc

```diff
@@ -71,6 +71,14 @@ Annotation& Annotation::operator=(const Annotation& ann) {
   return *this;
 }
 
+void Annotation::changeEndpointServiceName(const std::string& service_name) {
+  if (endpoint_.valid()) {
+    Endpoint endpoint(endpoint_.value());
```

Comment:
The const version can call the non-cost version. Basically never use const_cast.